### PR TITLE
[ESQL] clean up date trunc tests

### PR DESCRIPTION
**Pull request by not-napoleon**
While working on https://github.com/elastic/elasticsearch/issues/110008 I discovered that the Date Trunc tests were only running in folding mode, because the interval types are marked as not representable. The correct way to test this is to set the `forceLiteral` flag for those fields, which will (as the name suggests) force them to be literals even in non-folding tests.  

Doing that turned up errors in the `evaluatorToString` tests, which I fixed.  There are two big changes here.  First, the second parameter to the evaluator is a `Rounding` instance, not the actual interval.  Since `Rounding` includes some information about the specific rounding in the toString results, I am just using a starts with matcher to validate the majority of the string, rather than trying to reconstruct the expected rounding string.  Second, passing in a literal null for the interval parameter folds the whole expression to null, and thus a completely different toString.  I added a clause in `AnyNullIsNull` to account for this.

While I was in there, I moved some specific test cases to a different file.  I know moving code is something we're trying to minimize right now, but this seemed worth it.  The tests in question do not depend on the parameters of the test case, but all methods in the class get run for every set of parameters.  This was causing these tests to be run many times with the same values, which bloats our test run time and test count.  Moving them to a distinct class means they'll only be executed once per test run.  I feel like this benefit outweighs the cost of git history complexity.

### CODE REVIEW

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateTruncNonParameterizedTests.java
Comment (costin) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateTruncNonParameterizedTests.java:1]: NonParameterized seems fairly specific and not very descriptive - how about DateTruncRoundingTests?

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateTruncNonParameterizedTests.java
Comment (not-napoleon) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateTruncNonParameterizedTests.java:1]: Done.  I've also added some java doc to help clarify why we have two test classes.

**Comment by not-napoleon** [comment]
@elasticmachine update branch

**Pull request merged**